lib/_stream_wrap.js

```diff
@@ -118,3 +129,58 @@ StreamWrap.prototype.write = function write(req, bufs) {
 
   return 0;
 };
+
+StreamWrap.prototype._enqueue = function enqueue(req) {
+  if (this._queue === null) {
+    this._queue = req;
+    req._prev = req;
+    req._next = req;
+    return;
+  }
+
+  req._next = this._queue._next;
+  this._queue._next._prev = req;
+  req._prev = this._queue;
+  this._queue._next = req;
+};
+
+StreamWrap.prototype._dequeue = function dequeue(req) {
+  const next = req._next;
+  const prev = req._prev;
+
+  if (next === null && prev === null)
+    return false;
+
+  req._next = null;
+  req._prev = null;
+
+  if (next === prev) {
+    this._queue = null;
+  } else {
+    prev._next = next;
+    next._prev = prev;
+
+    if (this._queue === req)
+      this._queue = next;
+  }
+
+  return true;
+};
+
+StreamWrap.prototype.close = function close(cb) {
+  var self = this;
```

Comment:
I am really sorry that I missed this in the last commit's comment. This also looks like a good candidate for `const`.